src/query/functions/binary/unless.go

```diff
@@ -63,15 +62,46 @@ func makeUnlessBlock(
 		return nil, err
 	}
 
-	if err = appendValuesAtIndices(lIds, lIter, builder); err != nil {
+	for index := 0; lIter.Next(); index++ {
+		if !rIter.Next() {
+			return nil, errRExhausted
+		}
+
+		lStep := lIter.Current()
+		lValues := lStep.Values()
+		rStep := rIter.Current()
+		rValues := rStep.Values()
+		for iterIndex, valueIndex := range indices {
+			if valueIndex >= 0 {
+				if !math.IsNaN(rValues[iterIndex]) {
```

Comment:
nit: Perhaps join these too?  i.e. `if valueIndex >= 0 && !math.IsNan(...) { ... }`

Reply:
Sure, sounds good; generally find it a bit cleaner when they're split up when they have simple conditions, but might be going a bit overboard haha